### DISCUSSION TITLE
XAMLIL	Unable to resolve suitable regular or attached property

### DISCUSSION THREAD

**Discussion by NirabhraDas013** [AvaloniaUI/Avalonia#6637] [Q&A] [answered]
Hi,
I'm having a problem when building this in VS2019. The start of my axaml file looks like this

```
<Window x:Class="MHLab.Patch.Launcher.Wpf.MainWindow"
        xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:vm="clr-namespace:DataGridExample.ViewModels;assembly=DataGridExample"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        xmlns:local="clr-namespace:MHLab.Patch.Launcher.Wpf"
        mc:Ignorable="d" WindowStyle="None"
        Title="PATCH - WPF Launcher Sample" Height="297" Width="500"
        MouseDown="DragMoveMouseDown" ResizeMode="NoResize" ContentRendered="Window_ContentRendered" WindowStartupLocation="CenterScreen" Icon="/icon.png">
  <Grid>
```

Basically I copied it over from a WPF project. 
When I try to build, I get the error:

`Error	XAMLIL	Unable to resolve suitable regular or attached property WindowStyle on type MHLab.Patch.Launcher.Wpf:MHLab.Patch.Launcher.Wpf.MainWindow Line 1, position 2.	MHLab.Patch.Launcher.Wpf`

I tried taking out the "WindowStyle" and it complains about the next property "MouseDown"

I am at a loss on what to do. Please help

**Comment by jp2masa** [comment, marked as answer]
Avalonia doesn't have the same properties, you have to change the following.

- `WindowStyle="None"` -> `SystemDecorations="None"`
- `MouseDown` -> `PointerPressed`
- `ResizeMode="NoResize"` -> `CanResize="False"`
- `ContentRendered` doesn't exist I think

**Reply by NirabhraDas013** [reply]
Thanks so much. Is there any documentation on which properties transform to which?

Also, is there any On-Load type property that I can use instead of `ContentRendered ` to make sure my controls are ready at load?

**Reply by jp2masa** [reply]
Not sure exactly what kind of loading you're looking for, but probably one of the following:

- `AttachedToLogicalTree`
- `AttachedToVisualTree` (I would try this one first)
- `LayoutUpdated`
- `TemplateApplied`

You can read the reference here: http://reference.avaloniaui.net/api/Avalonia.Controls/Window/#Events.

**Reply by NirabhraDas013** [reply]
Thank you SO much. That link would help a Lot. I'm really mostly fumbling through the stuff here. Just started to work with wpf for the first time and just heard of Avalonia yesterday.
